Testscripts/Linux/STOR-Large-Disk-CopyFile.sh

```diff
@@ -0,0 +1,162 @@
+        LogMsg "Listing directory: ls /mnt/"
+        ls /mnt/
+        rm -rf /mnt/*
+        LogMsg "Disk test completed for file copy on ${drive_name}1 with filesystem ${fs}."
+    else
+        LogErr "Copying 5G file for ${drive_name}1 with filesystem ${fs} failed"
+        SetTestStateFailed
+        exit 0
+    fi
+}
+
+# test wget file, wget one 5G file to /mnt which is mounted to disk
+function Test_Wget_File() {
+    file_basename=`basename $wget_path`
+    wget -O /mnt/$file_basename $wget_path
+    file_size=`curl -sI $wget_path | grep Content-Length | awk '{print $2}' | tr -d '\r'`
+    file_size1=`ls -l /mnt/$file_basename | awk '{ print $5}' | tr -d '\r'`
+    LogMsg "file_size before wget=$file_size"
+    LogMsg "file_size after wget=$file_size1"
+    if [[ $file_size = $file_size1 ]]; then
+        LogMsg "Drive wget to ${drive_name}1 with filesystem ${fs} successfully"
+    else
+        LogErr "Drive wget to ${drive_name}1 with filesystem ${fs} failed"
+        SetTestStateFailed
+        exit 0
+    fi
+    rm -rf /mnt/*
+}
+
+# Format the disk and create a file system, mount and create file on it.
+function Test_FileSystem_Copy() {
+    drive=$1
+    fs=$2
+    parted -s -- $drive mklabel gpt
+    parted -s -- $drive mkpart primary 64s -64s
+    if [ "$?" = "0" ]; then
+        sleep 5
+        wipefs -a "${drive_name}1"
+        mkfs.$fs  ${drive_name}1
+        if [ "$?" = "0" ]; then
+            LogMsg "mkfs.${fs}   ${drive_name}1 successful..."
+            mount ${drive_name}1 /mnt
+            if [ "$?" = "0" ]; then
+                LogMsg "Drive mounted successfully..."
+                # step 1: test for local copy file
+                LogMsg "Start to test local copy file"
+                Test_Local_CopyFile
+                # step 2: wget 5GB file to disk
+                LogMsg "Start to test wget file"
+                Test_Wget_File
+                # umount /mnt files
+                umount /mnt
+                if [ "$?" = "0" ]; then
+                      LogMsg "Drive unmounted successfully..."
+                fi
+            else
+               LogErr "Error in mounting drive..."
+               SetTestStateFailed
+            fi
+        else
+            LogErr "Error in creating file system ${fs}.."
+            SetTestStateFailed
+        fi
+    else
+        LogErr "Error in executing parted  ${drive_name}1 for ${fs}"
+        SetTestStateFailed
+    fi
+}
+
+# Check for call trace log
+chmod +x check_traces.sh
+./check_traces.sh &
+
+# Count the number of SCSI= and IDE= entries in constants
+disk_count=1
+for entry in $(cat ./constants.sh)
+do
+    # Convert to lower case
+    low_str="$(tr '[A-Z]' '[a-z' <<<"$entry")"
+
+    # does it start wtih ide or scsi
+    if [[ $low_str == ide* ]];
+    then
+        disk_count=$((disk_count+1))
+    fi
+
+    if [[ $low_str == scsi* ]];
+    then
+        disk_count=$((disk_count+1))
+    fi
+done
+
+LogMsg "constants disk count= $disk_count"
```

Comment:
Tests will be performed on $disk_count disks

Reply:
Updated to "Tests will be performed on $disk_count disks"